source/AI.cpp

```diff
@@ -144,8 +146,36 @@ namespace {
 		{
 			// If no direct jump route, or the target system has no
 			// fuel, perform a more elaborate refueling check.
-			const DistanceMap distance(ship, to);
-			return ShouldRefuel(ship, distance);
+			vector<const System *> destinations = {to};
+			if(ship.IsYours() && player.HasTravelPlan())
+				reverse_copy(player.TravelPlan().begin(), player.TravelPlan().end(), back_inserter(destinations));
```

Comment:
Would `destinations.insert(destinations.end(), player.TravelPlan().rbegin(), player.TravelPlan().rend())` work here?

Should be able to drop the `<algorithm>` include then too.

Reply:
I don't recall a `reverse_interator` syntax in vector::insert, but I'll test it.